tests/datadriven/__init__.py

```diff
@@ -194,6 +200,24 @@ def assertRaisesWith(self, exceptionType):
                 "exception of type {} not raised".format(
                     exceptionType.__name__))
 
+    def assertAlmostEqual(self, a, b, ndigits=7):
+        """
+        Assert that a and b are equal within ndigits digits
+        """
+        if round(a-b, ndigits) != 0:
```

Comment:
Clever, and perfect! 
Well... almost (caveat stackoverflow). Need to take `abs( a_number)` before applying `log10` to it, otherwise you error out on negative inputs. Thus:

```
import math

def round_to_sigdigs( a_number, sig=7):
    return round(a_number, sig-int(math.floor(math.log10(abs(a_number))))-1)
```
etc...